src/oneMean/oneMean.js

```diff
@@ -0,0 +1,61 @@
+import { dropTextFileOnTextArea, parseCsvVariableByCol } from "../util/csv.js";
+import StackedDotChart from "../util/stackeddotchart.js";
+
+export class OneMean {
+  constructor(OneMeanDiv) {
+    this.ele = {
+      csvTextArea: OneMeanDiv.querySelector("#csv-input"),
+      loadDataBtn: OneMeanDiv.querySelector("#load-data-btn"),
+      variableDropDown: OneMeanDiv.querySelector("#variable")
+    };
+    this.datasets = [{ label: "Data", backgroundColor: "#333333", data: [] }];
+    this.dataChart1 = new StackedDotChart(
+      OneMeanDiv.querySelector("#data-chart-1"),
+      [this.datasets[0]]
+    );
+    this.loadEventListener = () => {
+      this.ele.loadDataBtn.addEventListener("click", e => {
+        this.data = parseCsvVariableByCol(this.ele.csvTextArea.value);
+        this.loadVariablesToDropDown(this.data, this.ele.variableDropDown);
+        e.preventDefault();
+      });
+
+      dropTextFileOnTextArea(this.ele.csvTextArea);
+    };
+
+    this.ele.variableDropDown.addEventListener("change", e => {
+      const varName = e.target.value;
+      if (e.target.value) this.updateData(varName, this.data);
+    });
+
+    this.loadEventListener();
+  }
+
+  loadVariablesToDropDown(dataSets, dropDownElement) {
+    const varNames = Object.keys(dataSets);
+    const html = varNames.reduce((acc, x) => {
+      return (acc += `<option value="${x}">${x}</option>`);
+    }, '<option value="null">Please Select Variable</option>');
+
+    dropDownElement.innerHTML = html;
+  }
+
+  updateData(varName, data) {
+    this.dataChart1.updateLabelName(0, varName);
+    this.dataChart1.setDataFromRaw([data[varName]]);
+    this.dataChart1.changeDotAppearance(1, undefined);
```

Comment:
Does this make the points one pixel wide?

Reply:
Yes, because  some one mean dataset are so big, some over 10000 pts